models/aether_lat-lon/model_mod.f90

```diff
@@ -0,0 +1,616 @@
+
+! Restart data is ordered (lev,lat,lon) (translated from C to fortran).
+call get_model_variable_indices(index_in, lev_index, lat_index, lon_index, &
+                                var_id=my_var_id, kind_index=my_qty)
+
+! should be set to the actual location using set_location()
+location = set_location(lons(lon_index), lats(lat_index), levs(lev_index), VERTISHEIGHT)
+
+! should be set to the physical quantity, e.g. QTY_TEMPERATURE
+if (present(qty)) qty = my_qty
+
+end subroutine get_state_meta_data
+
+!-----------------------------------------------------------------------
+! Does any shutdown and clean-up needed for model. Can be a NULL
+! INTERFACE if the model has no need to clean up storage, etc.
+
+subroutine end_model()
+
+end subroutine end_model
+
+!-----------------------------------------------------------------------
+! write any additional attributes to the output and diagnostic files
+
+subroutine nc_write_model_atts(ncid, domain_id)
+
+integer, intent(in) :: ncid      ! netCDF file identifier
+integer, intent(in) :: domain_id
+
+character(len=*), parameter :: routine = 'nc_write_model_atts'
+
+if ( .not. module_initialized ) call static_init_model
+
+! It is already in define mode from nc_create_file.
+! OR NOT, if called by create_and_open_state_output
+call nc_begin_define_mode(ncid)
+
+! nc_write_model_atts is called by create_and_open_state_output,
+!   which calls nf90_enddef before it.
+call nc_add_global_creation_time(ncid, routine)
+
+call nc_add_global_attribute(ncid, "model_source", source, routine)
+call nc_add_global_attribute(ncid, "model", "aether", routine)
+
+call nc_end_define_mode(ncid)
+
+! Flush the buffer and leave netCDF file open
+call nc_synchronize_file(ncid)
+
+end subroutine nc_write_model_atts
+
+!-----------------------------------------------------------------------
+! Read dimension information from the template file and use 
+! it to assign values to variables.
+
+subroutine assign_dimensions()
+
+integer  :: ncid
+character(len=24), parameter :: routine = 'assign_dimensions'
+
+call error_handler(E_MSG, routine, 'reading filter input ['//trim(template_file)//']')
+
+ncid = nc_open_file_readonly(template_file, routine)
+
+! levels
+nlev = nc_get_dimension_size(ncid, trim(LEV_DIM_NAME), routine)
+allocate(levs(nlev))
+call nc_get_variable(ncid, trim(LEV_VAR_NAME), levs, routine)
+
+! latitiude
+nlat = nc_get_dimension_size(ncid, trim(LAT_DIM_NAME), routine)
+allocate(lats(nlat))
+call nc_get_variable(ncid, trim(LAT_VAR_NAME), lats, routine)
+
+! longitude
+nlon = nc_get_dimension_size(ncid, trim(LON_DIM_NAME), routine)
+allocate(lons(nlon))
+call nc_get_variable(ncid, trim(LON_VAR_NAME), lons, routine)
+
+end subroutine assign_dimensions
+
+!-----------------------------------------------------------------------
+! Parse the table of variables' characteristics into arrays for easier access.
+
+function assign_var(variables, MAX_STATE_VARIABLES) result(var)
```

Comment:
nitpick: the indenting for function assign_var does not match the rest of the file.